src/Microsoft.Azure.SignalR.Management/IServiceManager.cs

```diff
@@ -38,9 +41,38 @@ public interface IServiceManager
         /// Creates an client endpoint for SignalR hub connections to connect to Azure SignalR Service
         /// </summary>
         /// <param name="hubName">The hub name.</param>
-        /// <returns>Client endpoint to Azure SignalR Service.</returns>
+        /// <returns>Client endpoint URL to Azure SignalR Service.</returns>
         string GetClientEndpoint(string hubName);
 
+
+        /// <summary>
+        /// Selects an endpoint randomly with <see cref="IEndpointRouter"/>
+        /// </summary>
+        /// <param name="httpContext"></param>
+        /// <returns>Client endpoint object to Azure SignalR Service.</returns>
+        ServiceEndpoint GetClientEndpoint(HttpContext httpContext);
+
+        /// <summary>
+        /// Gets the URL of a specified client endpoint for SignalR hub connections to connect to Azure SignalR Service.
+        /// </summary>
+        /// <param name="hubName">The hub name.</param>
+        /// <param name="endpoint">The endpoint object.</param>
+        /// <returns>Client endpoint URL to Azure SingalR Service</returns>
+        string GetClientEndpoint(string hubName, ServiceEndpoint endpoint);
+
+        /// <summary>
+        /// Creates a client access token for SignalR hub connections to connect to Azure SignalR Service.
+        /// </summary>
+        /// <param name="hubName">The hub name.</param>
+        /// <param name="endpoint">The client endpoint.</param>
+        /// <param name="userId">The user ID.</param>
+        /// <param name="claims">The claim list to be put into access token.</param>
+        /// <param name="lifeTime">The lifetime of the token. The default value is one hour.</param>
+        /// <returns>Client access token to Azure SignalR Service.</returns>
+        string GenerateClientAccessToken(string hubName, ServiceEndpoint endpoint, string userId = null, IList<Claim> claims = null, TimeSpan? lifeTime = null);
+
+
```

Comment:
extra lines

Reply:
I'd like to use extra blank lines to separate methods for single-endpoint mode and multiple-endpoint mode?